### PR TITLE
add phan as static code anaylser

### PR DESCRIPTION
**Pull request by patrickjahns**
## Motivation

With https://github.com/owncloud/core/pull/31807 it was a POC for using phan in core for detecting bugs / problems

This PR introduces phan with a lenient configuration to an app and shows how it needs to be configured for testing.

Once this is merged, the next steps include in making the checking stricter and adressing the issues found by phan

Edit: note that i've added quite a few people as reviewers to draw attention and discuss it

### CODE REVIEW

**Comment by phil-davis** [comment]
This sort of thing seems good to me. The settings must be very lenient, since you did not have to change any code to make it pass! It will be nice to enforce all sorts of crud that is easily missed if you ignore your IDE - unused vars, undefined vars, missing or wrong phpdoc...

**Comment by patrickjahns** [comment]
@phil-davis  - it's the most lenient setting - same here for core: https://github.com/owncloud/core/pull/31807  - where a lot of bad imports / missing imports have been caught.

increasing the strength - I find already 100 errors for this app - so we need to gradually work here. 
First introduce - ✅ - strengthen checks - fix 💪

**Comment by jvillafanez** [comment]
Nothing against this.

File: .phan/config.php [hunk 1]
```diff
@@ -0,0 +1,94 @@
+<?php
+
+/**
+ * This configuration will be read and overlaid on top of the
+ * default configuration. Command line arguments will be applied
+ * after this file is read.
+ */
+
+return [
```
Comment (PVince81) [inline .phan/config.php]: should we put this in a common location like the coding standards repo ?

Comment (patrickjahns) [reply .phan/config.php]: I would not suggest this - as these settings are depending on the layout of the app (where will code be). And the settings are definitely not the same for every app. Some apps we can have very strict settings fast, whereas some apps will require more lenient settings by design

**Comment by tomneedham** [comment]
👍

**Pull request merged**